File: leetcode/3033. Modify the Matrix.cpp
class Solution {
public:
    vector<vector<int>> modifiedMatrix(vector<vector<int>>& matrix) {
        for (int c = 0; c < matrix[0].size(); ++c) {
            int colMax = -1;
            for (int r = 0; r < matrix.size(); ++r) {
                colMax = max(colMax, matrix[r][c]);
            }

            for (int r = 0; r < matrix.size(); ++r) {
                if (matrix[r][c] == -1)
                    matrix[r][c] = colMax;
            }
        }
        return matrix;
    }
};

File: leetcode/3033. Modify the Matrix.py
class Solution:
    def modifiedMatrix(self, matrix: List[List[int]]) -> List[List[int]]:
        for c in range(len(matrix[0])):
            col_max = -1
            for r in range(len(matrix)):
                col_max = max(col_max, matrix[r][c])

            for r in range(len(matrix)):
                if matrix[r][c] == -1:
                    matrix[r][c] = col_max

        return matrix